### ISSUE TITLE
Binary messages are unsupported by the js pubsub API

### ISSUE THREAD

**Issue by roberttod** [GoogleCloudPlatform/gcloud-node#987]
Messages pulled from pubsub using the subscription API are converted from base 64 to UTF 8
https://github.com/GoogleCloudPlatform/gcloud-node/blob/master/lib/pubsub/subscription.js#L298

This is fine for messages that are serialized to strings but for binary data round trips to/from UTF 8 doesn’t work http://haacked.com/archive/2012/01/30/hazards-of-converting-binary-data-to-a-string.aspx/

If there was an encoding option that defaults to `utf8`, receiving binary data e.g. protobuf data from a java app would be possible (currently I am having to use a fork to get this working).

I would recommend an option like so

```js
const base64Subscription = pubsub.subscription('foo' { encoding: 'base64' })
base64Subscription.on('message', onBase64Message)

function onBase64Message (m) {
  // m.data is now a base 64 string rather than UTF 8
  const base64String = m.data
}
```

**Comment by stephenplusplus** [comment]
Thanks for reporting this. That sounds like a logical way to deal with this to me. Any chance you're willing to throw a PR together?

**Comment by roberttod** [comment]
Sure, I am just working on it now.

**Comment by stephenplusplus** [comment]
Fixed in #988 - thanks again!
